runtime/gc_realtime/RealtimeMarkingSchemeRootClearer.hpp

```diff
@@ -0,0 +1,291 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Metronome
+ */
+
+#if !defined(REALTIMEROOTCLEARER_HPP_)
+#define REALTIMEROOTCLEARER_HPP_
+
+#include "j9.h"
+#include "j9cfg.h"
+#include "modronopt.h"
+
+#include "EnvironmentRealtime.hpp"
+#include "RealtimeRootScanner.hpp"
+#include "RealtimeGC.hpp"
+#include "RealtimeMarkingScheme.hpp"
+
+/**
+ * This scanner will mark objects that pass through its doSlot.
+ */
+class MM_RealtimeMarkingSchemeRootClearer : public MM_RealtimeRootScanner
+{
+/* Data memebers / types */
+public:
+protected:
+private:
+	
+/* Methods */
+public:
+
+	/**
+	 * Simple chained constructor.
+	 */
+	MM_RealtimeMarkingSchemeRootClearer(MM_EnvironmentRealtime *env, MM_RealtimeGC *realtimeGC) :
+		MM_RealtimeRootScanner(env, realtimeGC)
+	{
+		_typeId = __FUNCTION__;
+	}
+	
+	/**
+	 * This should not be called
+	 */
+	MMINLINE virtual void
```

Comment:
Declaration `MMINLINE` is ignored with `virtual` and can be removed... here and everywhere